File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder sb=new StringBuilder();
        String s="";
        int di=0,small=0; 
        int l1=word1.length();
        int l2=word2.length();
        if(l2>l1)
        {
            di=l2-l1;
            small=l1;
            s=word2.substring(l1,l2);
        }
          if(l1>l2)
        {
          di=l1-l2;
          small=l2;
          s=word1.substring(l2,l1);
        }
        else
        {
            small=l1;
        }
        for(int i=0;i<small;i++)
        {
            sb.append(word1.charAt(i));
            sb.append(word2.charAt(i));
        }
        if(l1!=l2)
        sb.append(s);
        return sb.toString();
    }
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        l1=len(word1)
        l2=len(word2)
        s=""
        ans=""
        if l1-l2>0:
            d=l2
            s=word1[l2:]
        elif l2-l1>0:
            d=l1
            s=word2[l1:]
        else:
            d=l2

        for i in range(d):
            ans=ans+word1[i]+word2[i]  

        ans=ans+s  
        return ans   

